addons/survey/models/survey_survey.py

```diff
@@ -568,6 +568,32 @@ def _get_number_of_attempts_lefts(self, partner, email, invite_token):
     # QUESTIONS MANAGEMENT
     # ------------------------------------------------------------
 
+    def _remove_invalid_conditional_questions(self, questions):
```

Comment:
This method is not exactly "removing" records from the existing "questions" object, right? (which is good).
So maybe it should instead be called something like `_get_questions_to_show` , `_get_answerable_questions`  or `_get_questions_except_when_misplaced_trigger`.

Also, this should come lower in the class code, definitely below the single calling method.